app/models/npc.rb

```diff
@@ -0,0 +1,51 @@
+class Npc < ActiveRecord::Base
+  class ConsequenceLimit < ActiveModel::Validator
+    def validate(npc)
+      if npc.consequences.length > 4
+        npc.errors[:consequence] << 'Cannot have more than 4 consequences'
+      end
+    end
+  end
+
+  validates :name, :npc_type, :system_id, presence: true
+  validates_uniqueness_of :name, scope: :system
+  validates :max_physical_stress, :max_mental_stress, inclusion: 1..5
+  validates_with ConsequenceLimit
+  belongs_to :system
+  has_many :character_skills, dependent: :destroy
+  has_many :stunts, dependent: :destroy
+  has_many :aspects, as: :aspectable
+  has_many :consequences, before_add: :consequence_limit_validation
+
+  def consequence_limit
+    ConsequenceLimit.new
```

Comment:
To answer your original question: in this particular case I believe it is to only be called once, during validations. That being the case, do you think I should still memoize it just in case, or leave it as it is?